src/libcollections/vec.rs

```diff
@@ -1388,59 +1386,48 @@ impl<T: Clone> SpecFromElem for T {
     }
 }
 
-impl SpecFromElem for u8 {
-    #[inline]
-    fn from_elem(elem: u8, n: usize) -> Vec<u8> {
-        if elem == 0 {
+unsafe fn chunked_or<T, U: ops::BitOr<Output = U> + Copy>(x: T) -> U {
+    let p = &x as *const T as *const U;
+    let len = mem::size_of::<T>() / mem::size_of::<U>();
+    slice::from_raw_parts(p, len).iter().fold(mem::zeroed(), |state, &x| state | x)
+}
+
+fn is_zero<T: Copy>(x: T) -> bool {
+    unsafe {
+        match mem::align_of::<T>() {
+            n if n % 16 == 0 => 0u128 == chunked_or(x),
+            n if n % 8 == 0 => 0u64 == chunked_or(x),
+            n if n % 4 == 0 => 0u32 == chunked_or(x),
+            n if n % 2 == 0 => 0u16 == chunked_or(x),
+            _ => 0u8 == chunked_or(x),
+        }
+    }
+}
```

Comment:
Can you write some comments explaining how `chunked_or` and `is_zero` work? Are there any pitfalls? What happens when the alignment of a type is bigger than `16`?